rdt-app/src/main/java/io/ona/rdt_app/interactor/PatientRegisterActivityInteractor.java

```diff
@@ -0,0 +1,56 @@
+package io.ona.rdt_app.interactor;
+
+import com.vijay.jsonwizard.utils.FormUtils;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.smartregister.util.JsonFormUtils;
+
+import io.ona.rdt_app.callback.OnFormSavedCallback;
+import io.ona.rdt_app.model.Patient;
+
+import static io.ona.rdt_app.util.Constants.CONDITIONAL_SAVE;
+import static io.ona.rdt_app.util.Constants.ENCOUNTER_TYPE;
+import static io.ona.rdt_app.util.Constants.ENTITY_ID;
+import static io.ona.rdt_app.util.Constants.PATIENT_NAME;
+import static io.ona.rdt_app.util.Constants.PATIENT_REGISTRATION;
+import static io.ona.rdt_app.util.Constants.SEX;
+import static org.smartregister.util.JsonFormUtils.KEY;
+import static org.smartregister.util.JsonFormUtils.VALUE;
+import static org.smartregister.util.JsonFormUtils.getString;
+
+/**
+ * Created by Vincent Karuri on 05/08/2019
+ */
+public class PatientRegisterActivityInteractor {
+    /**
+     * Get the patient for whom the RDT is to be conducted
+     *
+     * @param jsonFormObject The patient form JSON
+     * @return the initialized Patient if proceeding to RDT capture otherwise return null patient
+     * @throws JSONException
+     */
+    public Patient getPatientForRDT(JSONObject jsonFormObject) throws JSONException {
+        Patient rdtPatient = null;
+        if (PATIENT_REGISTRATION.equals(jsonFormObject.optString(ENCOUNTER_TYPE))) {
+            JSONArray formFields = JsonFormUtils.fields(jsonFormObject);
+            JSONObject fieldJsonObject;
+            for (int i = 0; i < formFields.length(); i++) {
+                fieldJsonObject = formFields.getJSONObject(i);
+                if (CONDITIONAL_SAVE.equals(fieldJsonObject.optString(KEY)) &&
```

Comment:
What is the purpose of looping through the fields while the `fieldJsonObject` is not used. The field used to populate the patient are not dependent on `fieldJsonObject`.

If testing if the field exists why not use  `FormUtils.getFieldJSONObject(formFields, CONDITIONAL_SAVE)` check if its not null and value is 1

Reply:
That works. Done.